sdk/digitaltwins/azure-digitaltwins-core/API design.md

```diff
@@ -771,12 +771,55 @@ public DigitalTwinsResponse<Void> updateComponentWithResponse(String digitalTwin
 </details>
 
 ## Query
-<details>
-<summary><b>APIs</b></summary>
+<details><summary><b>Async APIs</b></summary>
+
+  ```java
+  /**
+  * Query digital twins.
+  *
+  * @param query The query string, in SQL-like syntax.
+  * @return A {@link PagedFlux} of application/json for the query result items.
+  */
+  @ServiceMethod(returns = ReturnType.COLLECTION)
+  public PagedFlux<String> query(String query)
+
+  /**
+  * Query digital twins.
+  *
+  * @param query The query string, in SQL-like syntax.
+  * @param clazz The model class to convert the query response to.
+  * @param <T> The generic type to convert the query response to.
+  * @return A {@link PagedFlux} of application/json for the query result items.
+  */
+  @ServiceMethod(returns = ReturnType.COLLECTION)
+  public <T> PagedFlux<T> query(String query, Class<T> clazz)
+  ```
+</details>
 
+<details><summary><b>Sync APIs</b></summary>
 
 ```java
-TODO:
+/**
+* Query digital twins.
+*
+* @param query The query string, in SQL-like syntax.
+* @param context Additional context that is passed through the Http pipeline during the service call.
+* @return A {@link PagedIterable} of application/json for the query result items.
+*/
+@ServiceMethod(returns = ReturnType.COLLECTION)
+public PagedIterable<String> query(String query, Context context)
+
+/**
+* Query digital twins.
+*
+* @param query The query string, in SQL-like syntax.
+* @param context Additional context that is passed through the Http pipeline during the service call.
+* @param clazz The model class to convert the query response to.
+* @param <T> The generic type to convert the query response to.
+* @return A {@link PagedIterable} of application/json for the query result items.
+*/
+@ServiceMethod(returns = ReturnType.COLLECTION)
+public <T> PagedIterable<T> query(String query, Class<T> clazz, Context context)
```

Comment:
Don't you need withResponse?

Reply:
No this does not return a response. This returns a pageable object to iterate on. The calls are make when you iterate on it.